src/main/webapp/app/core/util/alert.service.ts

```diff
@@ -1,38 +1,114 @@
-import { Injectable, SecurityContext, NgZone } from '@angular/core';
+import { Injectable, NgZone, SecurityContext } from '@angular/core';
 import { DomSanitizer } from '@angular/platform-browser';
 import { TranslateService } from '@ngx-translate/core';
 import { translationNotFoundMessage } from 'app/core/config/translation.config';
+import { EventManager, EventWithContent } from 'app/core/util/event-manager.service';
+import { AlertError } from 'app/shared/alert/alert-error.model';
+import { Subscription } from 'rxjs';
 
 export type AlertType = 'success' | 'danger' | 'warning' | 'info';
 
-export interface Alert {
-    id?: number;
+export interface AlertBase {
     type: AlertType;
     message?: string;
     translationKey?: string;
     translationParams?: { [key: string]: unknown };
     timeout?: number;
-    toast?: boolean;
-    position?: string;
-    close?: (alerts: Alert[]) => void;
+    action?: { label: string; callback: (alert: Alert) => void };
+    onClose?: (alert: Alert) => void;
+    dismissible?: boolean;
+}
+
+export interface Alert extends AlertBase {
+    id: number;
+    close: () => void;
```

Comment:
The close() function gets overwritten in the ``addAlert`` method, so we have a standard implementation for this that is always used. It's just convenient to be able to call ``alert.close()`` on an alert object returned by the alert service when calling ``addAlert``. Let me think about it, maybe I'll just make use of a class instead of a bunch of interfaces here ...

Reply:
I've updated the interface defintions to make the intentions more clear. Any attempt to overwrite ``close()`` after alert has been added will be prevented by TypeScript now. What do you think?